### ISSUE TITLE
Namespace member lookup on a separate file fails if import{} is used

### ISSUE THREAD

**Issue by fat4eyes-mwo** [Microsoft/TypeScript#26459]
**TypeScript Version:**  Tried on Version 3.1.0-dev.20180813, and 3.0.1, failed on both.

**Search Terms:** typescript import namespace lookup

**Code**
```ts
//file aa.ts
namespace A {
	export class Foo {
		constructor() {
		}
	}
	export function bar() {
	}
}
```
```ts
//file b.d.ts
export interface B {}
```
```ts
//file a-err.ts
import {B} from "./b"; //this import seems to cause lookup of A.foo and A.bar to fail below
namespace A {
	export class AA {
		constructor() {
		}
	}
	export function error() {
		//false errors, unable to find Foo and bar defined in aa.ts. 
		//Seems to be caused by the type import above.
		let a = new A.Foo();
		let b = A.bar();
		
		let c : B; //lookup ok
		let d = new A.AA(); //lookup ok
	}
}
```
```ts
//file a.ts
namespace A {
	export function main() {
		//works as expected, picks up Foo and bar defined in aa.ts
		let a = new A.Foo();
		let b = A.bar();
	}
}
```

**Expected behavior:** Lookup in a-err.ts should still be able to find class Foo and function bar defined (in the same namespace) in the separate file aa.ts

**Actual behavior:** Gives the following errors:
```
ts/import-err/a-err.ts:13:17 - error TS2339: Property 'Foo' does not exist on type 'typeof A'.

13      let a = new A.Foo();
                      ~~~

ts/import-err/a-err.ts:14:13 - error TS2339: Property 'bar' does not exist on type 'typeof A'.

14      let b = A.bar();
                  ~~~
```
Lookup does still work on classes defined in the same file.

**Playground Link:** n/a, playground doesn't do multiple files. Full test case can be found here: https://github.com/fat4eyes-mwo/ts-tests (ts files are in ts/import-err)

**Related Issues:** https://github.com/Microsoft/TypeScript/issues/23041 (could be related, but doesn't use import {})

**Comment by andy-ms** [comment]
When a file contains an `import`, that puts everything in it in the module's scope. Just like how a `const` in a module is separate from a global variable, a `namespace` in a module is local to that module. There's no good way to write to a global namespace from within a module.

**Comment by fat4eyes-mwo** [comment]
I see. So a file becomes a module the moment it uses an import.

As a note to myself and others, I've used this workaround to 'access' aa.ts from a-err.ts:

```ts
//in file a-err.ts
declare namespace A {
	export class Foo {
		constructor();
	}
	export function bar() : void;
}
```

It's basically the contents of the d.ts file for aa.ts copied into the file that used the import. 

This issue came up when I was trying to use Vue and typescript together without using any bundlers. I use global namespaces and old-fashioned <script> tags to pull in the generated .js files on the browser (my projects are very small). I had to do a

```ts
import { VueConstructor } from "vue/types/vue";
```

so I can get types on the Vue objects, which unfortunately made my source file unable to access my other global namespace files. I couldn't use an import statement to access my other files because it causes ts to emit an import statement in the generated .js (which is still not supported on browsers). Not a scalable workaround (fortunately I only needed a small set of the global namespace), but worked for my particular situation.

**Comment by andy-ms** [comment]
If you just need access to some types globally, you could declare a `globals.d.ts` file like so:
```ts
import { VueConstructor as ImportVueConstructor } from "vue/types/vue";
declare global {
    type VueConstructor = ImportVueConstructor;
}
```
Then you can access `VueConstructor` from any file.

**Comment by fat4eyes-mwo** [comment]
Wow that's much better. Thanks for the replies.
